hibernate-core/src/test/java/org/hibernate/orm/test/query/criteria/internal/hhh14916/Author.java

```diff
@@ -21,5 +22,6 @@ public class Author {
 	public String name;
 
 	@OneToMany(fetch = FetchType.LAZY, mappedBy = "author", orphanRemoval = true, cascade = CascadeType.ALL)
+	@OrderColumn
```

Comment:
Instead of requiring this change, I'd much rather distinguish this `LIST` association from a "real bag". Inverse collections (using `mappedBy`) can never allow duplicates, so we should actually always do deduplication and not treat these associations like bags.